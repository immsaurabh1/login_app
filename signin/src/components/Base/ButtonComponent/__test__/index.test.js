import React from "react";
import { render, cleanup } from "@testing-library/react";
import Button from "../index.js";
import "@testing-library/jest-dom/extend-expect";
import ReactDOM from "react-dom";

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders button correctly", () => {
  const { getByTestId } = render(<Button id="test" title="Login" />);
  expect(getByTestId("btn-cmp")).toHaveTextContent("Login");
});

it("matches snapshot", () => {
  const { container, getByTestId } = render(<Button id="test" title="Login" />);
  expect(getByTestId("btn-cmp")).toHaveTextContent("Login");
  expect(container).toMatchSnapshot();
});
