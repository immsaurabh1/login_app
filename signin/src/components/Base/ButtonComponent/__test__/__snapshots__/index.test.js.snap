// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    data-testid="btn-cmp"
    id="test"
    style="padding: 8px; font-size: 16px;"
    title="Login"
  >
    Login
  </button>
</div>
`;
